### PR TITLE
(fix): this fixes proguard rules when running minifyEnabled

### PR DESCRIPTION
**Pull request by thomaszurkan-optimizely**
## Summary
I was able to reproduce this problem turning on minify on a kotlin project.  The following rules and compile cleanups, enabled the apk to be created, installed, and ran, without issue. 

## Test plan
Created test apps to reproduce the issue.
 
## Issues
- #258

### CODE REVIEW

**Comment by thomaszurkan-optimizely** [comment]
@mnoman9 can you share your project with me? How did you install the aar s?

**Comment by thomaszurkan-optimizely** [comment]
@mnoman09 after looking at your project and changing it to use my branch instead of master for the fix, it worked.

**Comment by mnoman09** [comment]
yes I tested it with my project and It's working so lgtm.

File: shared/src/main/java/com/optimizely/ab/android/shared/ServiceScheduler.java [hunk 1]
```diff
@@ -157,11 +157,11 @@ private void cancelRepeating(PendingIntent pendingIntent, Intent intent) {
             Integer id = null;
             try {
                 id = (Integer) Class.forName(clazz).getDeclaredField("JOB_ID").get(null);
-                pendingIntent.cancel();
                 // only cancel periodic services
                 if (ServiceScheduler.isScheduled(context, id)) {
                     jobScheduler.cancel(id);
                 }
+                pendingIntent.cancel();
```
Comment (mikeng13) [inline shared/src/main/java/com/optimizely/ab/android/shared/ServiceScheduler.java]: What does moving this here do?

File: android-sdk/build.gradle [hunk 2]
```diff
@@ -66,8 +66,7 @@ dependencies {
         exclude group: 'com.google.code.findbugs'
     }
 
-    compileOnly group: 'org.slf4j', name: 'slf4j-android', version: '1.7.25'
```
Comment (mikeng13) [inline android-sdk/build.gradle]: We don't need this anymore?

File: android-sdk/build.gradle [hunk 3]
```diff
@@ -66,8 +66,7 @@ dependencies {
...
-
+    compileOnly "com.fasterxml.jackson.core:jackson-databind:$jacksonversion"
```
Comment (mikeng13) [inline android-sdk/build.gradle]: Does this mean we explicitly bundle jackson with our SDK? Would the user be able to exclude it from their app?

File: shared/src/main/java/com/optimizely/ab/android/shared/ServiceScheduler.java (hunk 1 above)
Comment (thomaszurkan-optimizely) [reply shared/src/main/java/com/optimizely/ab/android/shared/ServiceScheduler.java]: It is just consistent with what is below it.  We first cancel the alarm and then the pendingIntent.  Here we cancel the job and then the pending intent.  Basically, it is so if the pendingIntent throws, we have still cancelled the scheduled job.

File: android-sdk/build.gradle (hunk 2 above)
Comment (thomaszurkan-optimizely) [reply android-sdk/build.gradle]: We have a compile only for noveo android logger.  We don't need both.

File: android-sdk/build.gradle (hunk 3 above)
Comment (thomaszurkan-optimizely) [reply android-sdk/build.gradle]: It is compile only.  So, it is not a explicit dependency.  If you are not using jackson, then, there is no effect.  If you are using jackson, you would add this as a implementation in your gradle.  Essentially, this is so our jackson code compiles.  We still default to gson.

**Pull request merged**